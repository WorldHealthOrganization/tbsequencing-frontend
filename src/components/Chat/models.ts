export interface IUserMessage {
  email: string,
  username: string,
  firstName: string,
  lastName: string
}
