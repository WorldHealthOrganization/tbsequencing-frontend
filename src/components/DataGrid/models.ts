export interface IColumn<T> {
  name: keyof T;
  header: string;
}

export enum GridType {
  Gene = 'gene',
  Drug = 'drug',
}

export enum ScrollType {
  Infinite = 'infinite',
  Regular = 'regular',
}

export type SortOrder = 'asc' | 'desc' | undefined;

export interface ISortState {
  colIndex?: number
  order?: SortOrder
  colID?: string
}

export interface IColumnFilter {
  id: string;
  value: number | string;
}

export interface BaseColumns {
  variantName: string;
  variantGrade: string;
  consequence: string;
  globalFrequency: string;
  totalCounts: number;
  resistantCount: number;
  susceptbleCount: number;
  intermediateCount: number;
  nucleodicAnnName: string;
  proteicAnnName: string;
  drug: number;
  drugName: string;
}

export interface DrugTableColumns extends BaseColumns {
  geneName: string;
}

export interface GeneTableColumns extends BaseColumns {
  drug: number;
}
