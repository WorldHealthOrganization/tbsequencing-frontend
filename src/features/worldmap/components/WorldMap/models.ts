export type MapType = 'macroRegions' | 'heatmap';
