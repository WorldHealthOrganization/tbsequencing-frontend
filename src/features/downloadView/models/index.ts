import { IDrug } from '../../../services/drugsApi/models';

export interface IExtendedDrug extends IDrug {
  isChecked: boolean;
}
