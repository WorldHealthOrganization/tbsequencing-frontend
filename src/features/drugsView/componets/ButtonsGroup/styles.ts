import { css } from '@emotion/react';

const wrapper = css({
  marginBottom: '24px',
  display: 'flex',
});

const block = css({
  marginRight: '16px',
  display: 'flex',
  button: {
    whiteSpace: 'nowrap',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
  },
});

export {
  wrapper, block,
};
