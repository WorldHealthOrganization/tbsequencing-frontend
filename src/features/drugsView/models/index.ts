export enum DrugStatus {
  Susceptible = 'susceptible',
  Resistant = 'resistant',
  Intermediate = 'intermediate',
  RatioSusceptible = 'ratioSusceptible',
  RatioResistant = 'ratioResistant',
  RatioIntermediate = 'ratioIntermediate',
}

export enum ResistantType {
  Phenotypic = 'Pheno',
  Genotypic = 'Geno',
}

export enum ChartDataType {
  Ratio = 'ratio',
  Count = 'count',
}

export enum Regions {
  ALL = 'ALL_REGIONS',
  AFRICA = 'AFR',
  EUROPE = 'EUR',
  AMERICA = 'AMR',
  EASTERN_MEDITERRANEAN = 'EMR',
  WESTERN_PACIFIC = 'WPR',
  SOUTH_EAST_ASIA = 'SEA',
}
