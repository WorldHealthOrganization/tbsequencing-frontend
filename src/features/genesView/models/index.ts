import { skipToken } from '@reduxjs/toolkit/query';
import React from 'react';
import { IGene, IVariant } from '../../../services/genesApi/models';

export type AutocompleteParams = typeof skipToken | { search?: string };

export interface IGenomeContextWrapperProps {
  data?: IGene[];
  isLoading?: boolean;
  selectedGene?: IGenesOverview;
  minDomain?: number;
  maxDomain?: number;
}

export interface IGenomeContextProps extends IGenomeContextWrapperProps {
  width: number;
}

export interface IVariantLocal extends IVariant {
  pos: number;
}

export interface ISearchResult {
  geneName: string;
  geneDbCrossrefId: number;
  locusTag: string;
}

export interface ISearchCount extends ISearchResult {
  counter: number;
}

export interface ISearchRecent extends ISearchResult {
  counter: string;
  date: Date | string;
}

export interface IGenesOverview extends IGene {
  geneDescription: string;
  geneType: string;
  ncbiId: string;
  proteinLength: number;
}

export interface IGenesOverviewTableData {
  title: string;
  value: string | number | null;
  element?: React.ReactNode;
}

export interface IGenesOverviewResponse {
  genes: IGene;
  genesOverview: IGenesOverview
}
