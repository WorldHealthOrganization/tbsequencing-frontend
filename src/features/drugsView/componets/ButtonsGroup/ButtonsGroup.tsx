/** @jsxImportSource @emotion/react */
import React, { useCallback } from 'react';
import { AppButton } from '../../../../components/AppButton/AppButton';
import { block, wrapper } from './styles';
import { ChartDataType, Regions, ResistantType } from '../../models';
import AppSelectBox from '../../../../components/AppSelectBox';
// import RangeSlider from '../../../../components/RangeSlider';
import AppRadioGroup from '../../../../components/AppRadioGroup';
import { IDropDownContent } from '../../../../components/AppSelectBox/AppSelectBox';

interface IButtonGroupProps {
  resistantTypeClick: (type: ResistantType) => void;
  handleDataTypeChanged: (value: ChartDataType) => void;
  resistantType: ResistantType;
  chartDataType: ChartDataType,
  // setSelectedDates: (value: number[]) => void;
  handleRegionChange: (value: Regions) => void;
  selectedRegion: Regions;
  // selectedDates: number[];
}

const items = [
  { value: Regions.ALL, label: 'Global' },
  { value: Regions.AFRICA, label: 'Africa' },
  { value: Regions.AMERICA, label: 'America' },
  { value: Regions.EUROPE, label: 'Europe' },
  { value: Regions.EASTERN_MEDITERRANEAN, label: 'Eastern Mediterranean' },
  { value: Regions.SOUTH_EAST_ASIA, label: 'South-East Asia' },
  { value: Regions.WESTERN_PACIFIC, label: 'Western Pacific' },
];

// const now = new Date();

const ButtonsGroup = ({
  chartDataType,
  resistantType,
  resistantTypeClick,
  // setSelectedDates,
  handleRegionChange,
  handleDataTypeChanged,
  selectedRegion,
  // selectedDates,
}: IButtonGroupProps) => {
  const renderRadioGroup = useCallback(({ closeDropDown }: IDropDownContent) => (
    <AppRadioGroup
      variant="singleCol"
      selectedValue={selectedRegion}
      onChange={(value) => {
        handleRegionChange(value);
        closeDropDown();
      }}
      items={items}
    />
  ), [handleRegionChange, selectedRegion]); // TODO remove this var?

  // const renderRangeSlider = useCallback(({ closeDropDown }: IDropDownContent) => (
  //   <RangeSlider
  //     min={1964}
  //     max={now.getFullYear()}
  //     defaultValue={selectedDates}
  //     onSubmit={(e) => {
  //       setSelectedDates(e);
  //       closeDropDown();
  //     }}
  //   />
  // ), [setSelectedDates, selectedDates]);

  return (
    <div css={wrapper}>
      <div css={block}>
        <AppButton
          onClick={() => resistantTypeClick(ResistantType.Phenotypic)}
          variant={resistantType === ResistantType.Phenotypic ? 'contained' : 'outlined'}
          size="medium"
        >
          Phenotypic resistance
        </AppButton>
        <AppButton
          onClick={() => resistantTypeClick(ResistantType.Genotypic)}
          variant={resistantType === ResistantType.Genotypic ? 'contained' : 'outlined'}
          size="medium"
        >
          Genotypic resistance
        </AppButton>
      </div>
      <div css={block}>
        <AppSelectBox
          endIconName="arrow_drop_down"
          dropdownContentRender={renderRadioGroup}
          size="medium"
          variant="outlined"
        >
          WHO region
        </AppSelectBox>
      </div>
      {/* <div css={block}>
        <AppSelectBox
          endIconName="arrow_drop_down"
          dropdownContentRender={renderRangeSlider}
          size="medium"
          variant="outlined"
        >
          Time period
        </AppSelectBox>
      </div> */}
      <div css={block}>
        <AppButton
          onClick={() => handleDataTypeChanged(ChartDataType.Ratio)}
          variant={chartDataType === ChartDataType.Ratio ? 'contained' : 'outlined'}
          size="medium"
        >
          Proportions
        </AppButton>
        <AppButton
          onClick={() => handleDataTypeChanged(ChartDataType.Count)}
          variant={chartDataType === ChartDataType.Count ? 'contained' : 'outlined'}
          size="medium"
        >
          Absolute numbers
        </AppButton>
      </div>
    </div>
  );
};

export default ButtonsGroup;
