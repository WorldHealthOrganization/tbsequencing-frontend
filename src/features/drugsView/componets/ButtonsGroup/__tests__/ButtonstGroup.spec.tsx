import React from 'react';
import {
  getTreeJSONWithRedux, render, screen, fireEvent,
} from '../../../../../utils/testUtils/jestUtils';
import ButtonsGroup from '../ButtonsGroup';
import { ChartDataType, Regions, ResistantType } from '../../../models';

describe('ButtonsGroup', () => {
  const onResistantTypeClick = jest.fn();
  const setSelectedDates = jest.fn();
  const handleRegionChange = jest.fn();
  const handleDataTypeChanged = jest.fn();
  const props = {
    chartDataType: ChartDataType.Ratio,
    resistantType: ResistantType.Genotypic,
    resistantTypeClick: onResistantTypeClick,
    setSelectedDates,
    handleRegionChange,
    handleDataTypeChanged,
    selectedRegion: Regions.ALL,
    selectedDates: [1995, 2015],
  };
  test('renders ButtonsGroup components without error', () => {
    render(
      <ButtonsGroup {...props} />,
    );
  });

  test('renders all buttons correctly', () => {
    render(
      <ButtonsGroup {...props} />,
    );

    screen.getByText('Phenotypic resistance');
    screen.getByText('Genotypic resistance');
    screen.getByText('WHO region');
    screen.getByText('Time period');
    screen.getByText('Proportions');
    screen.getByText('Absolute numbers');
  });

  test('calls correct callbacks Resistant Type', () => {
    render(
      <ButtonsGroup {...props} />,
    );

    const btn = screen.getByText('Phenotypic resistance');
    fireEvent.click(btn);

    expect(onResistantTypeClick).toHaveBeenCalledTimes(1);
  });

  test('calls correct callbacks Chat Data Type', () => {
    render(
      <ButtonsGroup {...props} />,
    );

    const btn = screen.getByText('Absolute numbers');
    fireEvent.click(btn);

    expect(handleDataTypeChanged).toHaveBeenCalledTimes(1);
  });

  test('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <ButtonsGroup {...props} />,
    );

    expect(tree).toMatchSnapshot();
  });
});
