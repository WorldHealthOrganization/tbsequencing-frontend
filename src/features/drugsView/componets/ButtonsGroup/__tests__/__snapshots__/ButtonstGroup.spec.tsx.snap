// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonsGroup Should render correctly 1`] = `
<div
  className="css-1667gqu-ButtonsGroup"
>
  <div
    className="css-1pim0p0-ButtonsGroup"
  >
    <button
      className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButtonBase-root css-sfprtt-MuiButtonBase-root-MuiButton-root-AppButton"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-3cwtjx-MuiTypography-root-BaseText"
      >
        Phenotypic resistance
      </p>
    </button>
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root css-1ve9vu2-MuiButtonBase-root-MuiButton-root-AppButton"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-3cwtjx-MuiTypography-root-BaseText"
      >
        Genotypic resistance
      </p>
    </button>
  </div>
  <div
    className="css-1pim0p0-ButtonsGroup"
  >
    <button
      className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButtonBase-root css-sfprtt-MuiButtonBase-root-MuiButton-root-AppButton"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-3cwtjx-MuiTypography-root-BaseText"
      >
        WHO region
      </p>
      <span
        className="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
      >
        <span
          aria-hidden={true}
          className="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
        >
          arrow_drop_down
        </span>
      </span>
    </button>
  </div>
  <div
    className="css-1pim0p0-ButtonsGroup"
  >
    <button
      className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButtonBase-root css-sfprtt-MuiButtonBase-root-MuiButton-root-AppButton"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-3cwtjx-MuiTypography-root-BaseText"
      >
        Time period
      </p>
      <span
        className="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
      >
        <span
          aria-hidden={true}
          className="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
        >
          arrow_drop_down
        </span>
      </span>
    </button>
  </div>
  <div
    className="css-1pim0p0-ButtonsGroup"
  >
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root css-1ve9vu2-MuiButtonBase-root-MuiButton-root-AppButton"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-3cwtjx-MuiTypography-root-BaseText"
      >
        Proportions
      </p>
    </button>
    <button
      className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButtonBase-root css-sfprtt-MuiButtonBase-root-MuiButton-root-AppButton"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-3cwtjx-MuiTypography-root-BaseText"
      >
        Absolute numbers
      </p>
    </button>
  </div>
</div>
`;
